dcgm/tests/test_dcgm.py

```diff
@@ -0,0 +1,69 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+from unittest import mock
+
+import pytest
+
+from datadog_checks.dcgm import DcgmCheck
+
+
+@pytest.fixture
+def instance():
+    return {"openmetrics_endpoint": "http://localhost:9400/metrics"}
+
+
+@pytest.fixture
+def check(instance):
+    return DcgmCheck('dcgm.', {}, [instance])
+
+
+@pytest.fixture()
+def mock_metrics():
+    f_name = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'metrics.txt')
+    with open(f_name, 'r') as f:
+        text_data = f.read()
+    with mock.patch(
+        'requests.get',
+        return_value=mock.MagicMock(
+            status_code=200, iter_lines=lambda **kwargs: text_data.split("\n"), headers={'Content-Type': "text/plain"}
+        ),
+    ):
+        yield
+
+
+def test_check(dd_run_check, aggregator, check, mock_metrics):
+    dd_run_check(check)
+
+    for expected_metric in METRICS:
+        aggregator.assert_metric(
+            name=f"dcgm.{expected_metric['name']}",
+        )
+
+
+def test_service_checks(dd_run_check, aggregator, check, mock_metrics):
+    dd_run_check(check)
+    aggregator.assert_service_check('dcgm.openmetrics.health', DcgmCheck.OK)
```

Comment:
You can combine this with the previous test.

Reply:
I've added the service check in with the metric check ✅